urllib3/response.py

```diff
@@ -327,9 +384,14 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.strict_content_length and
+                       self.length_remaining not in (0, None)):
+                        raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
```

Comment:
Cool, in that case we should write this down in a comment too for when I inevitably forget that. ;)